sycl/test-e2e/DeviceArchitecture/device_architecture_comparison_on_device_aot.cpp

```diff
@@ -25,172 +25,172 @@ int main() {
+         }).otherwise([&]() {});
+
+         // negative test if_architecture_is_ge - 1
+         if_architecture_is_ge<architecture::intel_gpu_pvc_vg>([&]() {
+         }).otherwise([&]() { result[20] = 1; });
+
+         // negative test if_architecture_is_ge - 2
+         if_architecture_is_ge<architecture::nvidia_gpu_sm_89>([&]() {
+         }).otherwise([&]() { result[21] = 1; });
+
+         // test else_if_architecture_is_ge - 1
+         if_architecture_is_ge<architecture::intel_gpu_pvc_vg>([&]() {})
+             .else_if_architecture_is_ge<architecture::intel_gpu_pvc>(
+                 [&]() { result[22] = 1; })
+             .otherwise([&]() {});
+
+         // test else_if_architecture_is_ge - 2
+         if_architecture_is_ge<architecture::intel_gpu_pvc_vg>([&]() {
+         }).else_if_architecture_is_ge<architecture::nvidia_gpu_sm_89>([&]() {})
+             .else_if_architecture_is_ge<architecture::intel_gpu_pvc>(
+                 [&]() { result[23] = 1; })
+             .otherwise([&]() {});
+
+         // test if_architecture_is_between
+         if_architecture_is_between<architecture::intel_gpu_skl,
+                                    architecture::intel_gpu_pvc_vg>([&]() {
+           result[24] = 1;
+         }).otherwise([&]() {});
+
+         // negative test if_architecture_is_between - 1
+         if_architecture_is_between<architecture::intel_gpu_dg1,
+                                    architecture::intel_gpu_dg2_g12>([&]() {
+         }).otherwise([&]() { result[25] = 1; });
+
+         // negative test if_architecture_is_between - 2
+         if_architecture_is_between<architecture::intel_gpu_dg1,
+                                    architecture::nvidia_gpu_sm_89>([&]() {
+         }).otherwise([&]() { result[26] = 1; });
+
+         // test else_if_architecture_is_between - 1
+         if_architecture_is_between<architecture::intel_gpu_dg1,
+                                    architecture::intel_gpu_dg2_g12>([&]() {})
+             .else_if_architecture_is_between<architecture::intel_gpu_skl,
+                                              architecture::intel_gpu_pvc>(
+                 [&]() { result[27] = 1; })
+             .otherwise([&]() {});
+
+         // test else_if_architecture_is_between - 2
+         if_architecture_is_between<architecture::intel_gpu_dg1,
+                                    architecture::intel_gpu_dg2_g12>([&]() {})
+             .else_if_architecture_is_between<architecture::intel_gpu_dg1,
+                                              architecture::nvidia_gpu_sm_89>(
+                 [&]() {})
+             .else_if_architecture_is_between<architecture::intel_gpu_skl,
+                                              architecture::intel_gpu_pvc>(
+                 [&]() { result[28] = 1; })
+             .otherwise([&]() {});
+
+         // if adding new test here, don't forget to increment result's index
+         // and value of N variable
+       });
+     }).wait();
```

Comment:
The test must fee `result` memory.

Reply:
https://github.com/intel/llvm/pull/16667